maestro-studio/web/src/ActionModal.tsx

```diff
@@ -0,0 +1,219 @@
+import { UIElement } from './models';
+import { motion } from 'framer-motion';
+import React, { useLayoutEffect, useMemo, useState } from 'react';
+import { CommandExample, getCommandExamples } from './commandExample';
+import { useHotkeys } from 'react-hotkeys-hook';
+import copy from 'copy-to-clipboard';
+import Fuse from 'fuse.js';
+
+const KeyPill = ({text}: {
+  text: string
+}) => {
+  return (
+    <span className="whitespace-nowrap flex items-center px-2 h-5 bg-slate-200 text-slate-400 font-semibold text-xs rounded">
+      {text}
+    </span>
+  )
+}
+
+const Shortcut = ({text, action}: {
+  text: string | string[]
+  action: string
+}) => {
+  const textArray: string[] = typeof text === 'string' ? [text] : text
+  return (
+    <div className="flex gap-1.5">
+      <div className="flex gap-1">
+        {textArray.map(key => (
+          <KeyPill text={key} />
+        ))}
+      </div>
+      <span className="text-slate-400 text-sm whitespace-nowrap">
+        {action}
+      </span>
+    </div>
+  )
+}
+
+const SearchBar = ({query, onQuery}: {
+  query: string
+  onQuery: (s: string) => void
+}) => {
+  return (
+    <div className="flex items-center bg-slate-50 border-b border-b-slate-200">
+      <input
+        className="outline-none py-5 pl-5 pr-5 flex-1 bg-transparent min-w-[175px]"
+        placeholder="Search commands"
+        autoFocus={true}
+        value={query}
+        onChange={e => onQuery(e.target.value)}
+      />
+      <div className="flex gap-x-3 gap-y-2 pr-5 flex-wrap py-5 justify-end">
+        <Shortcut text={['\u25B2', '\u25BC']} action={"Select"} />
+        <Shortcut text={'\u2318 C'} action={"Copy"} />
+        <Shortcut text={'\u2318 D'} action={"Docs"} />
```

Comment:
Nit: here and in other places we should replace the macOS meta key symbol with Linux and Windows alternatives for users on those platforms.

Reply:
Let's tackle as a follow up if needed